packages/flutter_tools/lib/src/macos/xcdevice.dart

```diff
@@ -549,7 +552,11 @@ class XCDevice {
           iosDeploy: _iosDeploy,
           iMobileDevice: _iMobileDevice,
           platform: globals.platform,
-        ));
+        );
+
+        iosDevice.devModeEnabled = devModeEnabled;
```

Comment:
I would add this as part of the constructor. Looks like you had it as part of the constructor but then removed it, maybe because of @hellohuanlin's comment https://github.com/flutter/flutter/pull/125710#discussion_r1191575331? I think what he was meaning for you to do was make it not-null and have it be required to be passed in.

@hellohuanlin correct me if I'm wrong

Reply:
yep